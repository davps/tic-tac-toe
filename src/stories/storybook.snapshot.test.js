import React from 'react';
import renderer from 'react-test-renderer';

import initStoryshots from '@storybook/addon-storyshots';

initStoryshots();
