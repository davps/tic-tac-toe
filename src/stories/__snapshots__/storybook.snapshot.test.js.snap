// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Board Game is not over yet 1`] = `
<div>
  <span>
    <button
      className="square pending square-0"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-1"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-2"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
    <br />
  </span>
  <span>
    <button
      className="square pending square-3"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-4"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-5"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
    <br />
  </span>
  <span>
    <button
      className="square pending square-6"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-7"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-8"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
  </span>
</div>
`;

exports[`Storyshots Board Game over! 1`] = `
<div>
  <span>
    <span
      className="square placed square-0"
    >
      X
    </span>
  </span>
  <span>
    <span
      className="square placed square-1"
    >
      X
    </span>
  </span>
  <span>
    <span
      className="square placed square-2"
    >
      X
    </span>
    <br />
  </span>
  <span>
    <button
      className="square pending square-3"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-4"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-5"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <br />
  </span>
  <span>
    <button
      className="square pending square-6"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-7"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
  </span>
  <span>
    <button
      className="square pending square-8"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
  </span>
</div>
`;

exports[`Storyshots Full, live game Play it here 1`] = `
<div>
  <h1>
    Tic tac toe
  </h1>
  <br />
  <span>
     Jacob (X) 
  </span>
  <br />
  <span>
     David (O) 
  </span>
  <div>
    <span>
      <button
        className="square pending square-0"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-1"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-2"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <br />
    </span>
    <span>
      <button
        className="square pending square-3"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-4"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-5"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <br />
    </span>
    <span>
      <button
        className="square pending square-6"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-7"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-8"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
  </div>
  <div>
    <span>
      Now is the turn of
      <strong>
        <span>
           Jacob (X) 
        </span>
      </strong>
    </span>
  </div>
  <span />
  <span />
</div>
`;

exports[`Storyshots Game With initialized values 1`] = `
<div>
  <h1>
    Tic tac toe
  </h1>
  <br />
  <span>
     Jacob (X) 
  </span>
  <br />
  <span>
     David (O) 
  </span>
  <div>
    <span>
      <button
        className="square pending square-0"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-1"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-2"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <br />
    </span>
    <span>
      <button
        className="square pending square-3"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-4"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-5"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <br />
    </span>
    <span>
      <button
        className="square pending square-6"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-7"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
    <span>
      <button
        className="square pending square-8"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </span>
  </div>
  <div>
    <span>
      Now is the turn of
      <strong>
        <span>
           Jacob (X) 
        </span>
      </strong>
    </span>
  </div>
  <span />
  <span />
</div>
`;

exports[`Storyshots PlayAgainButton Game is not over yet (so this component should be hidden)  1`] = `<span />`;

exports[`Storyshots PlayAgainButton Game over! 1`] = `
<span>
  <button
    className="play-again"
    onClick={[Function]}
    type="button"
  >
    Play again
  </button>
</span>
`;

exports[`Storyshots PlayerInfo PlayerInfo of  val=PLAYER_1 label=X name=Jacob 1`] = `
<span>
   Jacob (X) 
</span>
`;

exports[`Storyshots PlayerInfo PlayerInfo of  val=PLAYER_2 label=O name=David 1`] = `
<span>
   David (O) 
</span>
`;

exports[`Storyshots Square Pending to use (displayed as a button) 1`] = `
<button
  className="square pending "
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Storyshots Square Pending to use but game is over 1`] = `
<button
  className="square pending "
  disabled={true}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Storyshots Square used by PLAYER_1  (X ) 1`] = `
<span
  className="square placed "
>
  X
</span>
`;

exports[`Storyshots Square used by PLAYER_2  (O ) 1`] = `
<span
  className="square placed "
>
  O
</span>
`;

exports[`Storyshots WhoIsNextInfo Game is not over and the next player is PLAYER_1  1`] = `
<div>
  <span>
    Now is the turn of
    <strong>
      <span>
         Jacob (X) 
      </span>
    </strong>
  </span>
</div>
`;

exports[`Storyshots WhoIsNextInfo Game over! (so this component should be hidden) 1`] = `<div />`;

exports[`Storyshots WinnerInfo Game is NOT over yet (so this component should be hidden) 1`] = `<span />`;

exports[`Storyshots WinnerInfo Game over! But without a winner, the board is full 1`] = `
<span>
  <div
    className="has-no-winner"
  >
    Nobody wins!
  </div>
</span>
`;

exports[`Storyshots WinnerInfo Game over! PLAYER_1 wins.  1`] = `
<span>
  <div
    className="has-winner"
  >
    The winner
     
    <strong>
      <span>
         Jacob (X) 
      </span>
    </strong>
  </div>
</span>
`;
